eng/Analyzers_NonShippingRules.ruleset

```diff
@@ -1,40 +0,0 @@
-<?xml version="1.0" encoding="utf-8"?>
-<RuleSet Name="Common diagnostic rules for all non-shipping projects" Description="Enables/disable rules specific to all non-shipping projects." ToolsVersion="14.0">
-  <!-- Include all shipping Roslyn-Analyzers project build rules -->
-  <Include Path=".\Analyzers_ShippingRules.ruleset" Action="Default" />
-
-  <Rules AnalyzerId="Microsoft.ApiDesignGuidelines.Analyzers" RuleNamespace="Microsoft.ApiDesignGuidelines.Analyzers">
-    <!-- Do not declare static members on generic types (we use generic static classes in test helpers) -->
-    <Rule Id="CA1000" Action="None" />
-    <!-- Nested types should not be visible (we use nested types as test helpers) -->
-    <Rule Id="CA1034" Action="None" />
-    <!-- IdentifiersShouldNotContainUnderscores - test names contain underscores and this is fine -->
-    <Rule Id="CA1707" Action="None" />
```

Comment:
This rule looks violated.

https://github.com/dotnet/roslyn-analyzers/blob/8a0442045d6d0a12a18851ffc93a920a5916d752/src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/Maintainability/AvoidUninstantiatedInternalClassesTests.cs#L21

I'm going to merge latest main since I recently updated the dogfooding analyzers version, just to confirm whether the build still passes or not.

Reply:
Build passed - something seems busted. I am going to create a test PR on main to check if at least the analyzers are running on main branch as expected.